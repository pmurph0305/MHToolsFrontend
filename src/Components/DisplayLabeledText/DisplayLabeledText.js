import React from "react";
import PropTypes from "prop-types";

import "./DisplayLabeledText.scss";
/**
 * @param  {string} label Label displayed above text
 * @param  {string} text  Text to be displayed
 */
const DisplayLabeledText = ({ label, text }) => {
  return (
    <div className="dlt-container">
      <label className="dlt-label">{label}</label>
      <p className="dlt-text">{text}</p>
    </div>
  );
};

DisplayLabeledText.propTypes = {
  label: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired
};

export default DisplayLabeledText;
