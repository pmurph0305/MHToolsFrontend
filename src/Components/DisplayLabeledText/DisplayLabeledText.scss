.dlt-container {
  padding-top: 1rem;
}

.dlt-label {
  font-size: 1.25rem;
  padding: 0.5rem;
  text-align: left;
}

.dlt-text {
  font-size: 1rem;
  text-align: left;
  padding: 0.5rem;
  line-height: 1.4rem;
}
