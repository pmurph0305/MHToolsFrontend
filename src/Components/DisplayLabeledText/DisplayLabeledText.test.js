import React from "react";
import { shallow } from "enzyme";
import DisplayLabeledText from "./DisplayLabeledText";

describe("DisplayLabeledText tests", () => {
  let wrapper;
  beforeEach(() => (wrapper = shallow(<DisplayLabeledText />)));

  it("Renders a div", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("Displays a label correctly", () => {
    wrapper.setProps({ label: "test" });
    expect(wrapper.text()).toEqual("test");
  });

  it("Displays text correctly", () => {
    wrapper.setProps({ text: "test text" });
    expect(wrapper.text()).toEqual("test text");
  });

  it("Displays both a label and text correctly", () => {
    wrapper.setProps({ label: "testLabel", text: "testText" });
    expect(wrapper.text()).toEqual("testLabeltestText");
  });

  it("Matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
