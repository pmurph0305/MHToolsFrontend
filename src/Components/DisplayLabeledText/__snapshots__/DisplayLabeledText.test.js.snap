// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayLabeledText tests Matches snapshot 1`] = `ShallowWrapper {}`;
